client/client.go

```diff
@@ -138,17 +133,17 @@ func (c *LogClient) UpdateSTR(ctx context.Context) error {
 	}
 
 	// Verify Consistency proof.
-	if str.TreeSize == c.str.TreeSize && bytes.Equal(str.RootHash, c.str.RootHash) {
+	if str.TreeSize == c.root.TreeSize && bytes.Equal(str.RootHash, c.root.RootHash) {
```

Comment:
Might want to just check the size here if log screws up and issues 2 STHs with different hashes we don't want to block here.